object-filter/src/main/java/org/infinispan/objectfilter/impl/aggregation/AvgAccumulator.java

```diff
@@ -0,0 +1,51 @@
+package org.infinispan.objectfilter.impl.aggregation;
+
+/**
+ * @author anistor@redhat.com
+ * @since 8.0
+ */
+public final class AvgAccumulator extends FieldAccumulator {
+
+   private static class Avg {
```

Comment:
Have you looked at using https://docs.oracle.com/javase/8/docs/api/java/util/DoubleSummaryStatistics.html ?  It does some special handling for averages that is probably more advanced then a simple division.